karibu-dsl/src/main/kotlin/com/github/mvysny/karibudsl/v10/Grid.kt

```diff
@@ -23,6 +24,27 @@ public inline fun <reified T : Any?> (@VaadinDsl HasComponents).grid(
         noinline block: (@VaadinDsl Grid<T>).() -> Unit = {}
 ): Grid<T> {
     val grid = Grid<T>(T::class.java, false)
+    return initGrid(grid, dataProvider, block)
```

Comment:
Gotcha. Could you rewrite your `treeGrid` fun as follows? `@VaadinDsl public fun <T: Any?> (@VaadinDsl HasComponents).treeGrid(klass: KClass<T>, ...)` ?

Reply:
@mvysny no, 'cause the `KClass` is defined in Kotlin as `interface KClass<T: Any>`.

You can't compile this:
```kotlin
fun <T: Any?> foobar(klass: KClass<T>) {}
```

just this:
```kotlin
fun <T: Any> foobar(klass: KClass<T>) {}
```

The error by IDEA:
![image](https://user-images.githubusercontent.com/6528869/110604939-e906bb80-8188-11eb-833d-cd73e60ce874.png)

Reply:
(we could rewrite this to use Java's `Class<T>` instead of the `KClass<T: Any>` of kotlin, and probably that should work with:
```kotlin
@VaadinDsl
public inline fun <reified T : Any?> (@VaadinDsl HasComponents).treeGrid(
        dataProvider: HierarchicalDataProvider<T, *>? = null,
        noinline block: (@VaadinDsl TreeGrid<T>).() -> Unit = {}
): TreeGrid<T> {
    return treeGrid(T::class.java, dataProvider, block)
}

@VaadinDsl
public fun <T : Any?> (@VaadinDsl HasComponents).treeGrid(
    clazz: Class<T>,
    dataProvider: HierarchicalDataProvider<T, *>? = null,
    block: (@VaadinDsl TreeGrid<T>).() -> Unit = {}
): TreeGrid<T> {
    val grid = TreeGrid(clazz)
    grid.removeAllColumns() // workaround for https://github.com/vaadin/vaadin-grid-flow/issues/973
    if (dataProvider != null) {
        (grid as Grid<T>).dataProvider = dataProvider
    }
    grid.hotfixMissingHeaderRow()
    return init(grid, block)
}
```

Reply:
Hmmm, I think having both treeGrid with `Class` parameter and `KClass` parameter would be beneficial. The same for `grid`. Could you please modify the PR in this way?

Reply:
@mvysny I've refactored the code, please review.

Reply:
Excellent work, thank you, looks good to me :+1: